yacc.go

```diff
@@ -3331,18 +3323,17 @@ type $$Parser interface {
 }
 
 type $$ParserImpl struct {
-	lookahead func() int
+	lval  $$SymType
+	stack [16]$$SymType
```

Comment:
You should keep the constant and just change `stacksize` to 16 (and maybe rename to `initialStackSize` since the original code seems to be under the impression that it's a limit rather than an initial value). A change that just changes the constant without making the other changes to `$$ParserImpl` might be more likely to be accepted upstream, if this is where a substantial portion of the performance gains come from.